# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1522089156 18000
#      Mon Mar 26 13:32:36 2018 -0500
# Branch stable
# Node ID 57d99b5e643a1487e74a2fbccd43469c91edbff1
# Parent  7feafd20703fab1053f92007dad5c93aefd87f71
Enable delete-on-close behavior of variable editor panels (bug #53467)

* variable-editor.cc (variable_dock_widget::variable_dock_widget): Drop the
  preprocessor conditional around the QA_DeleteOnClose setting.

diff --git a/libgui/src/variable-editor.cc b/libgui/src/variable-editor.cc
--- a/libgui/src/variable-editor.cc
+++ b/libgui/src/variable-editor.cc
@@ -83,24 +83,17 @@ make_plot_mapper (QMenu *menu)
 namespace octave
 {
   // Variable dock widget
 
   variable_dock_widget::variable_dock_widget (QWidget *p)
     : label_dock_widget (p)
   {
     setFocusPolicy (Qt::StrongFocus);
-    // This controls whether the variable_dock_widgets are deleted
-    // or hidden when clicking the upper right X icon.  (Look for
-    // the checkmark of the variable_editor's context menu.)
-    // Could make this an option, or perhaps add a drop-down menu
-    // in place of X "close" icon.
-#if 0
     setAttribute (Qt::WA_DeleteOnClose);
-#endif
 
     connect (m_dock_action, SIGNAL (triggered (bool)),
              this, SLOT (change_floating (bool)));
     connect (m_close_action, SIGNAL (triggered (bool)),
              this, SLOT (change_existence (bool)));
     connect (this, SIGNAL (topLevelChanged(bool)),
              this, SLOT (toplevel_change (bool)));
     connect (p, SIGNAL (visibilityChanged (bool)),
